plugins/module_utils/firewall_rules_utils.py

```diff
@@ -280,60 +282,122 @@ def __post_init__(self):
+            </source>
+        </rule>
+        ```
+
+        Note: The method assumes the presence of a utility function `dict_to_etree` for
+        converting dictionaries to XML elements.
+        """
+
         rule_dict: dict = asdict(self)
         del rule_dict["uuid"]
 
-        # Here data from the dataclass in this form eg.:
-        # { "source_any": "1" }
-        # into the required xml data structure:
-        # <source><any/></source>
         for direction in ["source", "destination"]:
             for key in ["address", "network", "port", "any", "not"]:
                 current_val: Optional[Any] = rule_dict.get(
                     f"{direction}_{key}"
                 )  # source_not = None
-                logging.info(f"{direction}_{key} : {current_val}")
                 if current_val is not None:
-
                     if rule_dict.get(direction) is None:
                         rule_dict[direction] = {}
 
                     # s/d_network
                     if isinstance(current_val, bool):
                         if current_val:
-                            # if current boolean value is 'True' it suffices
-                            # to write an empty tag into the xml:
-                            # e.g. {"any": True } == <any/>
-                            # no need for <any>1</any>
-
                             rule_dict[direction][key] = None
                     else:
-                        # every other option should be written into the
-                        # corresponding structure e.g:
-                        # { "source_port": 22 } == <source><port>22</port></source>
                         rule_dict[direction][key] = current_val
                 del rule_dict[f"{direction}_{key}"]
-        for rule_key, rule_val in rule_dict.copy().items():  # rule_val = "{"not":None}"
-            if rule_val is None or rule_val is False:
-                # remove unnecessary fields
+        for rule_key, rule_val in rule_dict.copy().items():
+            if (rule_val is None or rule_val is False) and rule_key != "quick":
                 del rule_dict[rule_key]
                 continue
             if issubclass(type(rule_val), ListEnum):
                 rule_dict[rule_key] = rule_val.value
+
+            elif rule_key == "quick":
+                if rule_val is True:
+                    del rule_dict["quick"]
+                else:
+                    rule_dict[rule_key] = "0"
+
             elif isinstance(rule_val, bool):
-                # assume it rule_val is True because of first check in the for loop
+                if "quick" in rule_key:
+                    raise Exception(rule_val)
                 rule_dict[rule_key] = "1"
 
-        # {"source":{"not":None}}
         element: Element = xml_utils.dict_to_etree("rule", rule_dict)[0]
 
         if self.uuid:
             element.attrib["uuid"] = self.uuid
 
         return element
 
+    # pylint: disable=too-many-locals
```

Comment:
We can avoid this lint rule by just simply combining declaration and dict-assignment like this:
```python3
rule_dict = {
    "interface": params.get("interface"),
    "type": params.get("action"),
    "descr": params.get("description"),
    "quick": params.get("quick"),
    "ipprotocol": params.get("ipprotocol"),
    "direction": params.get("direction"),
    "protocol": params.get("protocol"),
    "source_not": params.get("source_invert"),
    "source_address": params.get("source_ip") if params.get("source_ip") != "any" else None,
    "source_any": params.get("source_ip") is None or params.get("source_ip") == "any",
    "source_port": params.get("source_port"),
    "destination_not": params.get("target_invert"),
    "destination_address": params.get("target_ip") if params.get("target_ip") != "any" else None,
    "destination_any": params.get("target_ip") is None or params.get("target_ip") == "any",
    "destination_port": params.get("target_port"),
    "log": params.get("log"),
    "category": params.get("category"),
    "disabled": params.get("disabled"),
}
```

(again sorry for the missing suggestion, this code snippet is outside your diff...)

Reply:
thanks for the suggestion! Added in [6b30813](https://github.com/DonGiovanni83/puzzle.opnsense/pull/3/commits/6b30813801ff9977e4be018ac4bf8f8b90151ac7).